MasterServer/Program.cs

```diff
@@ -33,30 +33,30 @@ private static void Main(string[] args)
 				Console.WriteLine("Entering nothing will choose defaults.");
 				Console.WriteLine("Enter Host IP (Default: "+ GetLocalIPAddress() + "):");
 				read = Console.ReadLine();
-				if (string.IsNullOrEmpty(read))
-					host = GetLocalIPAddress();
-				else
-					host = read;
+				host = string.IsNullOrEmpty(read) ? GetLocalIPAddress() : read;
 
 				Console.WriteLine("Enter Port (Default: 15940):");
 				read = Console.ReadLine();
 				if (string.IsNullOrEmpty(read))
 					port = 15940;
 				else
-				{
 					ushort.TryParse(read, out port);
-				}
 			}
 
-			Console.WriteLine(string.Format("Hosting ip [{0}] on port [{1}]", host, port));
+			Console.WriteLine("Hosting ip [{0}] on port [{1}]", host, port);
 			PrintHelp();
 			MasterServer server = new MasterServer(host, port);
 			server.EloRange = eloRange;
 			server.ToggleLogging();
 
 			while (true)
 			{
-				read = Console.ReadLine().ToLower();
+				read = Console.ReadLine()?.ToLower();
+
+				// Running as a headless server
+				if (read == null)
```

Comment:
I would personally use a command line option to tell it to run as a daemon, maybe wit `-d` or something and would completely skipp the `ReadLine` call.

Reply:
Great idea! I've pushed some new commits for you to check out